arangod/Pregel/IncomingCache.cpp

```diff
@@ -215,12 +215,14 @@ void ArrayInCache<M>::forEach(
 // ================== CombiningIncomingCache ==================
 
 template<typename M>
-CombiningInCache<M>::CombiningInCache(std::set<PregelShard> localShards,
-                                      MessageFormat<M> const* format,
-                                      MessageCombiner<M> const* combiner)
-    : InCache<M>(format), _combiner(combiner), _localShards(localShards) {
+CombiningInCache<M>::CombiningInCache(
+    containers::FlatHashSet<PregelShard> localShards,
+    MessageFormat<M> const* format, MessageCombiner<M> const* combiner)
+    : InCache<M>(format),
+      _combiner(combiner),
+      _localShards(std::move(localShards)) {
   // one mutex per shard, we will see how this scales
-  for (PregelShard pregelShard : localShards) {
+  for (PregelShard const& pregelShard : localShards) {
```

Comment:
Must be changed here, because we moved out of `localShards` above.
```suggestion
  for (PregelShard const& pregelShard : _localShards) {
```

Reply:
And really at least the compiler should have told me about this :roll_eyes: